scripts/n-commits-since-last-tag.py

```diff
@@ -17,26 +18,42 @@ def warn_and_exit(message, exit_code=0):
     sys.exit(exit_code)
 
 
-if __name__ == "__main__":
+def run_subprocess(args, check=True):
     output = subprocess.run(
-        ["git", "--version"],
-        stderr=subprocess.DEVNULL,
-        stdout=subprocess.DEVNULL,
+        args,
+        capture_output=True,
+        encoding="utf8",
         check=False,
     )
 
-    if output.returncode != 0:
-        warn_and_exit("could not run `git`, is it installed on your system?")
+    if check and output.returncode != 0:
+        raise Exception(
+            f"failed to run '{' '.join(args)}' (exit code {output.returncode})\n"
+            f"stdout: {output.stdout}\n"
+            f"stderr: {output.stderr}\n"
+        )
+
+    if output.stderr != "":
+        print(output.stderr, file=sys.stderr)
+
+    return output
 
+
+if __name__ == "__main__":
     if len(sys.argv) != 2:
         warn_and_exit(f"usage: {sys.argv[0]} <tag-prefix>", exit_code=1)
 
-    result = subprocess.run(
-        ["git", "rev-parse", "--show-toplevel"],
-        capture_output=True,
-        check=False,
-        encoding="utf8",
-    )
+    result = run_subprocess(["git", "--version"], check=False)
+
+    if result.returncode != 0:
+        warn_and_exit("could not run `git`, is it installed on your system?")
+
+    # We need git >=2.0 for `git tag --sort=-creatordate` below
+    _, _, git_version, *_ = result.stdout.split()
```

Comment:
Wow, fancy dropping of return values. 😅

Reply:
Yeah, the issue is that on most Linux, git version is `git version 2.40`, but on macOS we get `git version 2.39.3 (Apple Git-145)`. This tries to handle both!

Reply:
Yes, I know these struggles. I played around with these things and then I decided to use versioneer for everything.

I hope this is not something which will be broken within every release of a new macOS version. 😅